Pull request for ref #61 and #62

arsp2701 opened the pull request:
See issues #62 and #61

doumdi commented:
@arsp2701 please add unit tests with the commit for verification.

doumdi commented on teraserver/python/modules/FlaskModule/API/user/UserQueryDevices.py:
```diff
@@ -49,6 +49,16 @@
+    def _value_counter(args):
+        res = 0
+        for value in args.values():
+            if value is None:
```
Fix to if value is not None:

doumdi replied:
remove res += 0

arsp2701 replied:
Changed

arsp2701 commented:
Added unit tests for the issues. Also changed the UserQueryDevices to fix the false boolean type issue (ie. false type returns nothing but goes in as 'not none')
The test_UserQueryDevices is for the UserQueryDevices.py file. It covers the 3 functions, get/post/delete. It has a flag randomizer for the 6 flags in order to test them all randomly.

SBriere commented on teraserver/python/modules/FlaskModule/API/user/UserQueryDevices.py:
```diff
@@ -67,37 +77,49 @@ def get(self):
 
-        if args['uuid']:
-            args['device_uuid'] = args['uuid']
+        # as_available = args.pop('available')
+        as_projects = args.pop('projects')
+        as_enabled = args.pop('enabled')
+        as_list = args.pop('list')
+        as_with_participants = args.pop('with_participants')
+        as_with_sites = args.pop('with_sites')
+        as_with_status = args.pop('with_status')
```
Should be "has_projects", "has_enabled", ... since otherwise it could cause confusion ("cast object as projects" instead of "object has projects").

arsp2701 replied:
Changed, thank you
